### PR TITLE
Upgrade python to 3.9.14 to fix CVE-2020-10735

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Python 3.9.14 is minor security update which fixes a few issues including CVE-2020-10735.

###### Change Log  

- Update Python to 3.9.14

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://discuss.python.org/t/python-versions-3-10-7-3-9-14-3-8-14-3-7-14-now-available/18844
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-10735
- https://nvd.nist.gov/vuln/detail/CVE-2020-10735 (Not registered with NIST yet)

###### Test Methodology

- Pipeline build in progress: 234195

### CODE REVIEW

**Review by mfrw — APPROVED** [review]
LGTM :)

File: toolkit/scripts/toolchain/container/toolchain-remote-wget-list [hunk 1]
```diff
@@ -20,7 +20,7 @@ https://invisible-mirror.net/archives/ncurses/ncurses-6.2.tar.gz
 http://ftp.gnu.org/gnu/patch/patch-2.7.6.tar.xz
 https://pkg-config.freedesktop.org/releases/pkg-config-0.29.2.tar.gz
 https://sourceforge.net/projects/procps-ng/files/Production/procps-ng-3.3.17.tar.xz
-https://www.python.org/ftp/python/3.9.13/Python-3.9.13.tar.xz
+https://www.python.org/ftp/python/3.9.14/Python-3.9.14.tar.xz
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain-remote-wget-list:23]: Revert unnecessarily changes to stage 1 toolchain.
```suggestion
https://www.python.org/ftp/python/3.9.13/Python-3.9.13.tar.xz
```

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 2]
```diff
@@ -777,9 +777,9 @@ popd
 rm -rf openssl-1.1.1k
 touch /logs/status_openssl_complete
 
-echo Python-3.9.13
-tar xf Python-3.9.13.tar.xz
-pushd Python-3.9.13
+echo Python-3.9.14
+tar xf Python-3.9.14.tar.xz
+pushd Python-3.9.14
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:782]: Revert unnecessarily changes to stage 1 toolchain.
```suggestion
echo Python-3.9.13
tar xf Python-3.9.13.tar.xz
pushd Python-3.9.13
```

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 3]
```diff
@@ -792,7 +792,7 @@ chmod -v 755 /usr/lib/libpython3.9.so.1.0
 chmod -v 755 /usr/lib/libpython3.so
 ln -sfv pip3.9 /usr/bin/pip3
 popd
-rm -rf Python-3.9.13
+rm -rf Python-3.9.14
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:795]: Revert unnecessarily changes to stage 1 toolchain.
```suggestion
rm -rf Python-3.9.13
```

File: toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh [hunk 4]
```diff
@@ -502,15 +502,15 @@ rm -rf perl-5.32.0
 
 touch $LFS/logs/temptoolchain/status_perl_complete
 
-echo Python-3.9.13
-tar xf Python-3.9.13.tar.xz
-pushd Python-3.9.13
+echo Python-3.9.14
+tar xf Python-3.9.14.tar.xz
+pushd Python-3.9.14
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh:507]:
```suggestion
echo Python-3.9.13
tar xf Python-3.9.13.tar.xz
pushd Python-3.9.13
```

File: toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh [hunk 5]
```diff
@@ -502,15 +502,15 @@ rm -rf perl-5.32.0
...
 sed -i '/def add_multiarch_paths/a \        return' setup.py
 ./configure --prefix=/tools --without-ensurepip --enable-shared
 make -j$(nproc)
 make install
 popd
-rm -rf Python-3.9.13
+rm -rf Python-3.9.14
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain_build_temp_tools.sh:513]:
```suggestion
rm -rf Python-3.9.13
```

File: toolkit/scripts/toolchain/container/toolchain-sha256sums [hunk 6]
```diff
@@ -47,7 +47,7 @@ ac610bda97abe0d9f6b7c963255a11dcb196c25e337c61f94e4778d632f1d8fd  patch-2.7.6.ta
 6fc69c01688c9458a57eb9a1664c9aba372ccda420a02bf4429fe610e7e7d591  pkg-config-0.29.2.tar.gz
 5159bc03a20b28ce363aa96765f37df99ea4d8850b1ece17d1e6ad5c24fdc5d1  popt-1.18.tar.gz
 4518b3e7aafd34ec07d0063d250fd474999b20b200218c3ae56f5d2113f141b4  procps-ng-3.3.17.tar.xz
-125b0c598f1e15d2aa65406e83f792df7d171cdf38c16803b149994316a3080f  Python-3.9.13.tar.xz
+651304d216c8203fe0adf1a80af472d8e92c3b0e0a7892222ae4d9f3ae4debcf  Python-3.9.14.tar.xz
```
Comment (dmcilvaney) [inline toolkit/scripts/toolchain/container/toolchain-sha256sums:50]:
```suggestion
125b0c598f1e15d2aa65406e83f792df7d171cdf38c16803b149994316a3080f  Python-3.9.13.tar.xz
```

**Comment by dmcilvaney** [comment]
Per chat with Andrew, better to leave the stage 1 toolchain static where possible. The stage 2 build is the important one to keep updated.

**Review by anphel31 — APPROVED** [review]

**Comment by mfrw** [comment]
> Per chat with Andrew, better to leave the stage 1 toolchain static where possible. The stage 2 build is the important one to keep updated.

@dmcilvaney: Should we document it somewhere ?
Probably in the the `stage1` place itself, so that we do not repeat this :)

**Pull request merged**